### PR TITLE
[vendor:directx/d3d_compiler] Added default include file handler

### PR DESCRIPTION
**Pull request by ftphikari**

### CODE REVIEW

File: vendor/directx/dxgi/dxgidebug.odin [hunk 1]
```diff
@@ -139,3 +139,21 @@ IDebug1_VTable :: struct {
 	DisableLeakTrackingForThread:   proc "stdcall" (this: ^IDebug1),
 	IsLeakTrackingEnabledForThread: proc "stdcall" (this: ^IDebug1) -> BOOL,
 }
+
+
+GetDebugInterface: proc "stdcall" (riid: ^IID, ppDebug: ^rawptr) -> HRESULT
+// Call this to start using GetDebugInterface
+debug_interface_init :: proc() -> bool {
```
Comment (gingerBill) [inline vendor/directx/dxgi/dxgidebug.odin:146]: Should this follow the same naming conventions as D3D? e.g. `InitDebugInterface`

File: vendor/directx/dxgi/dxgidebug.odin [hunk 2]
```diff
@@ -139,3 +139,21 @@ IDebug1_VTable :: struct {
...
+	if debug_lib == nil {
+		return false
+	}
+
+	ptr := win32.GetProcAddress(debug_lib, "DXGIGetDebugInterface")
+	if ptr == nil {
+		return false
+	}
+
+	GetDebugInterface = auto_cast(ptr)
```
Comment (gingerBill) [inline vendor/directx/dxgi/dxgidebug.odin:157]:
```odin
GetDebugInterface = auto_cast ptr
```

File: vendor/directx/dxgi/dxgidebug.odin (hunk 1 above)
Comment (ftphikari) [reply vendor/directx/dxgi/dxgidebug.odin:146]: No, it's just a helper function

Comment (gingerBill) [reply vendor/directx/dxgi/dxgidebug.odin:146]: But why not?

Comment (ftphikari) [reply vendor/directx/dxgi/dxgidebug.odin:149]: In GL package, helper functions do not follow GL convention, for example

Comment (ftphikari) [reply vendor/directx/dxgi/dxgidebug.odin:149]: Okay, I'll change it then

**Comment by ftphikari** [comment]
Is there anything preventing this from being merged?

**Comment by ftphikari** [comment]
To be honest I'm not even sure about the entire `GetDebugInterface` commit. `DXGIGetDebugInterface1` is available since Windows 8.1, and I think it's the rule of thumb in Odin that it does not support Windows lower than 8.1, right? So now I'm reconsidering even adding it in the first place, since it's only useful on Windows systems before 8.1.

**Review by gingerBill — APPROVED** [review]

**Pull request merged**